packages/core/test/buttons/buttonTests.tsx

```diff
@@ -84,5 +95,42 @@ function buttonTestSuite(component: React.ComponentClass<any>, tagName: string)
             const element = React.createElement(component, props);
             return useMount ? mount(element) : shallow(element);
         }
+
+        function checkClickTriggeredOnKeyUp(done: MochaDone,
+                                            buttonProps: Partial<IButtonProps>,
+                                            keyEventProps: Partial<React.KeyboardEvent<any>>) {
+            const wrapper = button(buttonProps, true);
+
+            // private members in TS are still accessible, and since Enzyme
+            // doesn't properly propagate events on non-string refs
+            // (https://github.com/airbnb/enzyme/issues/566), we need to dig in
+            // and attach a spy to the ref's .click function ourselves.
+            const buttonRef = (wrapper.instance() as any).buttonRef;
+            const onClick = sinon.spy(buttonRef, "click");
+
+            wrapper.simulate("keyup", keyEventProps);
+
+            // wait for the whole lifecycle to run
+            setTimeout(() => {
+                assert.equal(onClick.callCount, 1);
+                done();
+            }, 0);
+        }
+
+        function checkKeyEventCallbackInvoked(callbackPropName: string, eventName: string, keyCode: number) {
+            const callback = sinon.spy();
+
+            // IButtonProps doesn't include onKeyDown or onKeyUp in its
+            // definition, even though Buttons support those props. Casting as
+            // `any` gets around that for the purpose of these tests.
+            const wrapper = button({ [callbackPropName]: callback } as any);
```

Comment:
you can edit the `button` helper function to use `IButtonProps & ButtonHTMLProps` (I cant remember the exact nomenclature)